users/usersource.go

```diff
@@ -17,26 +19,39 @@ type StaticSource struct {
 	nextUser *ring.Ring
 }
 
+// NewStaticSource creates a StaticSource of all lower-cased and sorted from the specified users.
 func NewStaticSource(users ...string) *StaticSource {
+	for i, u := range users {
+		users[i] = strings.ToLower(u)
+	}
 	sort.Strings(users)
+
 	ss := &StaticSource{nextUser: ring.New(len(users))}
 	for _, u := range users {
-		ss.nextUser.Value = u
+		ss.nextUser.Value = strings.ToLower(u)
 		ss.nextUser = ss.nextUser.Next()
 	}
 	return ss
 }
 
-func (ss *StaticSource) StartAfter(user string) error {
+func (ss *StaticSource) StartAfter(user string) {
+	user = strings.ToLower(user)
+
+	var beginning *ring.Ring
 	for linksChecked := 0; linksChecked <= ss.nextUser.Len(); linksChecked++ {
-		if user == ss.nextUser.Prev().Value.(string) {
+		prevUser := ss.nextUser.Prev().Value.(string)
+		nextUser := ss.nextUser.Value.(string)
+		if nextUser < prevUser {
```

Comment:
Is this PR now adding a constraint that the static source of users needs to be sorted alphabetically?

I'm not sure I fully understand the bug that's being fixed here...is this addressing a case where we try to call `StartAfter` with a user who isn't in the rotation at all?  Is that a valid use case to support, or should we just return an error in that case?

Reply:
The list of users is and has always been sorted upon `StaticSource` creation in `NewStaticSource`.

> is this addressing a case where we try to call StartAfter with a user who isn't in the rotation at all?

Yes, this is the exact bug. It's an uncommon case where the user in the last shift (the only place where this method is called) is no longer in the rotation. Imagine if `duftler` had been assigned the last shift, and since he's no longer in the `build-cops` team, this would have throw an error saying it couldn't start the rotation after him. So without this change, a perfectly valid schedule generation when he was on the team would have caused an unnecessary error. 

I plan on addressing the purging of users no longer in the rotation in the next PR.